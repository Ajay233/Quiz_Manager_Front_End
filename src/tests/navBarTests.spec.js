import React from 'react'
import NavBar from '../navBar'
import DropdownList from '../dropdown/dropdownList'
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import history from '../history'

const mockStore = configureStore({})

describe("NavBar", () => {
  it("should render with just the options home and menu", () => {
    const store = mockStore({})

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <NavBar />
        </Router>
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })

  it("should render render the drop down whe menu button is clicked", () => {
    const store = mockStore({
      userData: { id: 1, loggedIn: true, permission: "ADMIN"}
    })

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <NavBar />
        </Router>
      </Provider>
    )

    wrapper.find('button').simulate('click')
    expect(wrapper.find(DropdownList)).toHaveLength(1)
  })

  it("should add an event listener when menu is clicked and remove it when clicked again", () => {
    const store = mockStore({
      userData: { id: 1, loggedIn: true, permission: "ADMIN"}
    })

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <NavBar />
        </Router>
      </Provider>
    )
    document.addEventListener = jest.fn()
    document.removeEventListener = jest.fn()

    const menu = wrapper.find('button').simulate('click')
    menu.simulate('click')
    expect(document.addEventListener).toHaveBeenCalledTimes(1)
    menu.simulate('click')
    expect(document.removeEventListener).toHaveBeenCalledTimes(1)
  })
})
