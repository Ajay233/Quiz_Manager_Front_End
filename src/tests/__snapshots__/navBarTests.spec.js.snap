// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render with just the options home and menu 1`] = `
<div>
  <div
    className="nav"
  >
    <a
      className="links"
      href="/"
      id="home"
      onClick={[Function]}
    >
      <i
        className="fas fa-home"
      />
       Home
    </a>
    <button
      className="linkButton links navItem"
      onClick={[Function]}
    >
      Menu 
      <i
        className="fas fa-bars"
      />
    </button>
    <span
      className="navItem"
    >
      <div
        className="toggle"
      >
        <i
          className="fas fa-yin-yang"
        />
         Toggle Theme
        <label
          className="switch"
        >
          <input
            id="darkModeToggle"
            onClick={[Function]}
            type="checkbox"
          />
          <span
            className="slider round"
          />
        </label>
      </div>
    </span>
  </div>
</div>
`;
